packages/wdio-browserstack-service/src/util.ts

```diff
@@ -97,7 +107,169 @@ export function getParentSuiteName(fullTitle: string, testSuiteTitle: string): s
+    }
+
+    Object.getOwnPropertyNames(prototype).forEach((methodName) => {
+        const method = prototype[methodName]
+        if (typeof method === 'function' && methodName !== 'constructor') {
+            // In order to preserve this context, need to define like this
+            Object.defineProperty(prototype, methodName, {
+                writable: true,
+                value: function(...args: any) {
+                    try {
+                        const result = method.call(this, ...args)
+                        if (result instanceof Promise) {
+                            return result.catch(error => processError(error, method, args))
+                        }
+                        return result
+
+                    } catch (err) {
+                        processError(err, method, args)
+                    }
+                }
+            })
+        }
+    })
+
+    return errorClass
+}
+
+export async function uploadCrashReport(exception: any, stackTrace: string) {
+    try {
+        if (!credentialsForCrashReportUpload.username || !credentialsForCrashReportUpload.password) {
+            credentialsForCrashReportUpload = process.env.CREDENTIALS_FOR_CRASH_REPORTING !== undefined ? JSON.parse(process.env.CREDENTIALS_FOR_CRASH_REPORTING) : credentialsForCrashReportUpload
+        }
+    } catch (error) {
+        return log.error(`[Crash_Report_Upload] Failed to parse user credentials while reporting crash due to ${error}`)
+    }
+    if (!credentialsForCrashReportUpload.username || !credentialsForCrashReportUpload.password) {
+        return log.error('[Crash_Report_Upload] Failed to parse user credentials while reporting crash')
+    }
+
+    try {
+        if (Object.keys(userConfigForReporting).length === 0) {
+            userConfigForReporting = process.env.USER_CONFIG_FOR_REPORTING !== undefined ? JSON.parse(process.env.USER_CONFIG_FOR_REPORTING) : {}
+        }
+    } catch (error) {
+        log.error(`[Crash_Report_Upload] Failed to parse user config while reporting crash due to ${error}`)
+    }
+
+    try {
+        const data = {
+            hashed_id: process.env.BS_TESTOPS_BUILD_HASHED_ID,
+            observability_version: {
+                frameworkName: 'WebdriverIO-' + (userConfigForReporting.framework || 'null'),
+                sdkVersion: bstackServiceVersion
+            },
+            exception: {
+                error: exception.toString(),
+                stackTrace: stackTrace
+            },
+            config: userConfigForReporting
+        }
+        const url = `${DATA_ENDPOINT}/api/v1/analytics`
+        const response: string = await got.post(url, {
+            ...DEFAULT_REQUEST_CONFIG,
+            ...credentialsForCrashReportUpload,
+            json: data
+        }).text()
+        log.debug(`[Crash_Report_Upload] Success response: ${JSON.stringify(response)}`)
+    } catch (error) {
+        log.error(`[Crash_Report_Upload] Failed due to ${error}`)
+    }
```

Comment:
no need for try/catch:
```suggestion
    const data = {
        hashed_id: process.env.BS_TESTOPS_BUILD_HASHED_ID,
        observability_version: {
            frameworkName: 'WebdriverIO-' + (userConfigForReporting.framework || 'null'),
            sdkVersion: bstackServiceVersion
        },
        exception: {
            error: exception.toString(),
            stackTrace: stackTrace
        },
        config: userConfigForReporting
    }
    const url = `${DATA_ENDPOINT}/api/v1/analytics`
    return got.post(url, {
        ...DEFAULT_REQUEST_CONFIG,
        ...credentialsForCrashReportUpload,
        json: data
    }).text().then(
        (response: string) => log.debug(`[Crash_Report_Upload] Success response: ${JSON.stringify(response)}`),
        (error) => log.error(`[Crash_Report_Upload] Failed due to ${error}`)
    )
```

Reply:
Done